EleksTubeHAX/Backlights.cpp

```diff
@@ -80,9 +81,35 @@ void Backlights::loop() {
   pattern_needs_init = false;
 }
 
-// TODO Implement these
-void Backlights::pulsePattern() {}
-void Backlights::breathPattern() {}
+void Backlights::pulsePattern() {
+  if (pattern_needs_init) {
+    fill(phaseToColor(config->color_phase));
+  }
+
+  float pulse_length_millis = (60.0f * 1000) / config->breath_per_min;
+  float val = 1 + abs(sin(2 * M_PI * millis() / pulse_length_millis)) * 254;
```

Comment:
I haven't built and tested this, but if feels like a linear, or even exponential ramp would be better than a `abs(sin( foo ))` here.  The problem is that a sine wave is changing its quickest closest to zero.  LEDs already have a much larger apparent change when the PWM values are closer to zero than when they're closer to `MAX_VAL`.  A roughly linear apparent brightness change can be had over `n` with `2**n`: 0, 1, 2, 4, 8, 16, 32, etc. `abs(sin( foo ))` is opposite, it changes QUICKER when its closer to zero.

I'm not going to reject the PR because of this, but I am curious what you think about it.